### PR TITLE
[CadbaseLibrary] Update to v0.2.0: add functionality for uploading files

### PR DESCRIPTION
**Pull request by mnnxp**
Thank you for creating a pull request to contribute to FreeCAD-macros!
To integrate your macro please make sure the following steps are complete:

- [ ] Please check this box if you're not submitting a new macro.   
- [ ] Are you submitting a new macro ?  
- [x] Have you followed the ['How to submit a macro'](../README.md#how-to-submit-a-macro) section of the README.md ?  
- [x] Your macro has a [Description](../README.md#macro-description) in its header.  
- [x] Your macro has a [CamelCase name](../README.md#camelcase-macro-name).  
- [x] Your macro is named [appropriately](../README.md#macro-name-specifics).  
- [x] Your macro contains a [Metadata section](../README.md#macro-metadata) that immediately follows the header description.  
- [x] Your macro is Python3/Qt5 compliant and tested on the latest FreeCAD stable and development releases.  
- [x] You're including documentation on how your macro works (bonus: screenshots and/or video on the Wiki)  
- [ ] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)  
- [x] Commit message is titled in the following way `[MacroName] Short description`.
- [x] Optional, write or update the changelog in the macro, from latest to oldest.

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

Sorry for the long message in the first commit.

A video on how this macro works (02:20): https://youtu.be/ONtNrYrg3HU

### CODE REVIEW

**Review by galou — APPROVED** [review]
Thanks for the contribution.

@mnnxp your macro starts to look more like a workbench. It is already too long for me to review. Can you please consider changing it into a workbench? According to the amount of functionalities it seems more appropriate. Thanks!

File: ImportExport/CadbaseLibrary/CdbsEvn.py [hunk 1]
```diff
@@ -1,33 +1,61 @@
-''' Global variables of the macro are collected here '''
+""" Global variables of the macro are collected here """
 
 import pathlib
-from PySide.QtGui import QFileDialog, QApplication
-import FreeCAD as app
+from PySide.QtGui import QFileDialog, QApplication  # FreeCAD's PySide
+import FreeCAD as App
```
[the remark this answers is not in the document]
Comment (mnnxp) [reply ImportExport/CadbaseLibrary/CdbsEvn.py:5]: Sorry I saw it in some macro or other place (maybe in IDE recommendations) and thought I should do it like this.
In order not to waste your time now, I will change this in a future update with other changes. But if I need to make it a separate PR now, please tell me.

**Comment by mnnxp** [comment]
@galou  thank you for the review!

Sorry if I made the wrong choice between macro and workbench.

I assumed that the interface of this macro could be run separately from the active workbench so that the user does not have to switch from the active workbench.
If this choice is wrong, then yes, I will consider how to change it.

**Comment by galou** [comment]
The choice is not wrong, there's no real difference between macro and workbench. You can also install global toolbars for workbenches. The advantage for you is that you won't need to go through the peer review.

As soon as you have your workbench ready (by the way, this will be in a separate repo), just write a PR to delete the macro (add a link to the new repo to the commit message then).

**Pull request merged**